src/Microsoft.OpenApi/Helpers/DictionaryCloneHelper.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.OpenApi.Helpers
+{
+    /// <summary>
+    /// Helper class for deep cloning dictionaries.
+    /// </summary>
+    public class DictionaryCloneHelper
```

Comment:
make the class and the method internal, we don't want consumers to take a dependency on that